subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -467,10 +497,21 @@ int nrf_cloud_fota_endpoint_set_and_report(struct mqtt_client *const client,
 		return ret;
 	}
 
-	ret = report_validated_job_status();
-	/* Positive value indicates no job exists, ignore */
-	if (ret > 0) {
-		ret = 0;
+	if ((current_fota.status == NRF_CLOUD_FOTA_IN_PROGRESS) &&
+	    (saved_job.validate == NRF_CLOUD_FOTA_VALIDATE_PENDING)) {
+		/* In case of a disconnection from nRF Cloud during the
+		 * FOTA update, send a job update to the cloud
+		 */
+		ret = send_job_update(&current_fota);
```

Comment:
If you've been disconnected, how can you send an update?

Or is this the DISCON message to tell a device it has been removed from nRF Cloud?

Reply:
this function (`nrf_cloud_fota_endpoint_set_and_report`) is called upon connection to the cloud.
suggestions improving the comment?

Reply:
"....FOTA update, queue a job update to the cloud to be sent on reconnection..."

Reply:
this function is called when (re)connected, the job update is sent immediately.
the comment is referring to a prior disconnect event.

Reply:
I think referring to the disconnection as prior would help a lot with the clarity of the comment

```c
		/* In case of a prior disconnection from nRF Cloud during the
		 * FOTA update, send a job update to the cloud
		 */
```